components/renku_data_services/crc/db.py

```diff
@@ -858,3 +866,72 @@ async def update_user(self, api_user: base_models.APIUser, keycloak_id: str, **k
             if (no_default_access := kwargs.get("no_default_access")) is not None:
                 user.no_default_access = no_default_access
             return user.dump()
+
+
+@dataclass
+class ClusterRepository:
+    """Repository for cluster descriptions."""
+
+    session_maker: Callable[..., AsyncSession]
+
+    @_only_admins
+    async def select_all(self, api_user: base_models.APIUser) -> list[Cluster]:
+        """Get cluster descriptions from the database."""
+
+        async with self.session_maker() as session:
+            r = await session.scalars(select(ClusterORM))
+            clusters = r.fetchall()
+
+            return [c.to_cluster() for c in clusters]
```

Comment:
suggestion: Make this an async iterator and use sqlalchemy streams. See for example `get_all_users` in `db.py` in the users component.

Reply:
done